plugins/modules/iam_user.py

```diff
@@ -227,44 +228,27 @@ def convert_friendly_names_to_arns(connection, module, policy_names):
         module.fail_json(msg="Couldn't find policy: " + str(e))
 
 
-def wait_iam_exists(connection, module):
-    if module.check_mode:
-        return
-    if not module.params.get('wait'):
-        return
-
-    user_name = module.params.get('name')
-    wait_timeout = module.params.get('wait_timeout')
-
-    delay = min(wait_timeout, 5)
-    max_attempts = wait_timeout // delay
-
-    try:
-        waiter = connection.get_waiter('user_exists')
-        waiter.wait(
-            WaiterConfig={'Delay': delay, 'MaxAttempts': max_attempts},
-            UserName=user_name,
-        )
-    except botocore.exceptions.WaiterError as e:
-        module.fail_json_aws(e, msg='Timeout while waiting on IAM user creation')
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg='Failed while waiting on IAM user creation')
-
-
-def create_or_update_login_profile(connection, module):
+def create_or_update_login_profile(connection, module, name):
 
     # Apply new password / update password for the user
     user_params = dict()
-    user_params['UserName'] = module.params.get('name')
+    user_params['UserName'] = name
     user_params['Password'] = module.params.get('password')
     user_params['PasswordResetRequired'] = module.params.get('password_reset_required')
     retval = {}
 
     try:
-        retval = connection.update_login_profile(**user_params)
+        retval = connection.update_login_profile(aws_retry=True, **user_params)
+        if module.params.get('wait'):
+            # Wait for login profile to finish updating
+            connection.update_login_profile(aws_retry=True, **user_params)
     except is_boto3_error_code('NoSuchEntity'):
+        # Login profile does not yet exist - create it
         try:
-            retval = connection.create_login_profile(**user_params)
+            retval = connection.create_login_profile(aws_retry=True, **user_params)
+            if module.params.get('wait'):
```

Comment:
Same thing here

Reply:
A waiter exists for create_login_profile, and this module has a helper that wraps it up in some extra ansible bits.  We should use that instead of repeatedly calling change methods.

Reply:
A waiter wasn't needed here so marking as resolved